src/lib.rs

```diff
@@ -90,41 +100,107 @@ pub struct IronOxide {
     pub(crate) rng: Mutex<ReseedingRng<ChaChaCore, EntropyRng>>,
 }
 
+/// Result of calling `initialize_check_rotation`
+pub enum InitAndRotationCheck {
+    /// Initialization succeeded, and no requests for private key rotations were present
+    NoRotationNeeded(IronOxide),
+    /// Initialization succeeded, but some keys should be rotated
+    RotationNeeded(IronOxide, PrivateKeyRotationCheckResult),
+}
+
+impl InitAndRotationCheck {
+    /// Caller asked to check rotation on initialize, but doesn't want to handle the result.
+    /// Consider using [initialize](fn.initialize.html) instead.
+    pub fn unwrap(self) -> IronOxide {
+        match self {
+            InitAndRotationCheck::NoRotationNeeded(io)
+            | InitAndRotationCheck::RotationNeeded(io, _) => io,
+        }
+    }
+}
+
+/// number of bytes that can be read from the RNG before it is reseeded. 1 MB
+const BYTES_BEFORE_RESEEDING: u64 = 1 * 1024 * 1024;
+
+/// Provides soft rotation capabilities for user and group keys
+pub struct PrivateKeyRotationCheckResult {
+    pub rotations_needed: EitherOrBoth<UserId, Vec<GroupId>>,
+}
+
+impl PrivateKeyRotationCheckResult {
+    pub fn user_rotation_needed(&self) -> Option<UserId> {
+        match &self.rotations_needed {
+            EitherOrBoth::Left(u) | EitherOrBoth::Both(u, _) => Some(u.clone()),
+            _ => None,
+        }
+    }
+
+    //    pub fn group_rotation_needed() -> Option<Vec<GroupId>> {
```

Comment:
I think we should remove these if you're not planning to deal with them now.

Reply:
I'm inclined to leave them here as we will be tackling this right away and otherwise we'd need to move the ideas here to tickets or elsewhere.

Reply:
I hate leaving things laying around the codebase (especially when we could shift any day). The code here doesn't feel like it really provides _that_ much value....